### ISSUE TITLE
Disable registration

### ISSUE THREAD

**Issue by Ale1x** [thedevdojo/auth#41]
Hello, is there a way to disable Users's registration?

By the way, great project, love it.

**Comment by tnylea** [comment]
Yeah, that could definitely be a feature. So, in the settings you were thinking about just including an extra toggle that says: `Disable Registration`, correct?

Where do you imagine the user would be redirected to when visiting `/auth/register` if it's disabled. Would the desirable redirect be to the `/auth/login` page?

**Comment by Ale1x** [comment]
I prefer that users be redirected to the ‘/’ route. If a site admin chooses to disable registration, it is likely that they do not want to disclose the existence of the login page either.

**Comment by johntrickett86** [comment]
+1 on disable user registration

I'd be happy with /auth/register just returning a 404 when disabled

**Comment by skdishansachin** [comment]
I want to work on this enhancement if it hasn't been assigned to anyone else. @tnylea

**Comment by skdishansachin** [comment]
Hey everyone, just a quick reminder to consider this: I'm currently looking into what to do when the user lands on `/auth/register`. How about adding an input box to the authentication setup for specifying the redirection path instead of hardcoding it? This way, you have more flexibility.

@Ale1x suggested redirecting to `/` and @johntrickett86 suggested redirecting to the `404` page. I'd appreciate some input on this. Thank you!

**Comment by johntrickett86** [comment]
> Hey everyone, just a quick reminder to consider this: I'm currently looking into what to do when the user lands on `/auth/register`. How about adding an input box to the authentication setup for specifying the redirection path instead of hardcoding it? This way, you have more flexibility.
> 
> @Ale1x suggested redirecting to `/` and @johntrickett86 suggested redirecting to the `404` page. I'd appreciate some input on this. Thank you!

Maybe there could be a bool setting of Redirect if disabled, then if true you would need to specify the path?

**Comment by skdishansachin** [comment]
@johntrickett86, can you please explain it more? I didn't quite get it.

**Comment by johntrickett86** [comment]
> @johntrickett86, can you please explain it more? I didn't quite get it.

So let's say that in the auth settings page there is an option (Boolean) to disable the registration. 

If false, then registration is allowed. 

If true, there are further settings:
1. Allow redirect (Boolean) - if false, registration route will return a 404 if true then registration route will redirect elsewhere
2. Redirect path: allows user to specify where the redirect path if registration is disabled and the allow redirect option is true

It could be potentially cleaner to simply have a redirect path option and if null return a 404.

**Comment by skdishansachin** [comment]
@johntrickett86 Okay now I understand it. That's a cleaner and better way to handle this thank you.

**Comment by tnylea** [comment]
@skdishansachin I haven't started implementing this so feel free to start implementing 😉 I've just assigned you to this, but let me know if you need any assistance. Really appreciate it!

I've merged your password confirmation into main and will be tagging a new version today 👏

**Comment by skdishansachin** [comment]
@tnylea I will start working on this soon. Thank you.

**Comment by skdishansachin** [comment]
Just wanted to give a quick update on this. I've been busy with personal stuff and haven't had a chance to start working on it yet. I'll get on it soon, though. I'm hoping to finish this before the month is up. I think this feature has some potential, so I wanted to keep you in the loop. Thanks for your patience!

**Comment by skdishansachin** [comment]
I could use some help with this. I'm trying to figure out how to put this together since it's a bit unclear to me. I'm doing what @johntrickett86 recommended -

> 1, Allow redirect (Boolean) - if false, registration route will return a 404 if true then registration route will redirect elsewhere
> 2, Redirect path: allows user to specify where the redirect path if registration is disabled and the allow redirect option is true

So, I was thinking about the registration options, like whether to include the name field, email verification, and password confirmation. I'm wondering what happens to those values when the registration is disabled. Do they just get hidden or turned off?

Also, I'm thinking about how to show the input field for the redirect path when registration is disabled. In my mind, when the disable registration is set to false by default, and the user turns it on, there should be a field to fill in the redirect path. This input box will only be visible if the registration is disabled.
